packages/merkle_tree/src/merkle_tree.cairo

```diff
@@ -129,18 +131,18 @@ pub impl MerkleTreeImpl<T, +HasherTrait<T>, +Copy<T>, +Drop<T>> of MerkleTreeTra
     fn verify(
         ref self: MerkleTree<T>, root: felt252, mut leaf: felt252, mut proof: Span<felt252>
     ) -> bool {
-        while let Option::Some(proof_element) = proof
-            .pop_front() {
-                // Compute the hash of the current node and the current element of the proof.
-                // We need to check if the current node is smaller than the current element of the proof.
-                // If it is, we need to swap the order of the hash.
-                leaf =
-                    if Into::<felt252, u256>::into(leaf) < (*proof_element).into() {
-                        self.hasher.hash(leaf, *proof_element)
-                    } else {
-                        self.hasher.hash(*proof_element, leaf)
-                    };
-            };
+        while let Option::Some(proof_element) = proof.pop_front() {
```

Comment:
Same as above, `for proof_element in proof`

Reply:
Done